### PR TITLE
Refactor image manager for client/server implementation of the container runtime

### PR DESCRIPTION
**Pull request by feiskyer**
Refactor image manager for client/server implementation of the container runtime at #13768

### CODE REVIEW

**Comment by feiskyer** [comment]
cc @dchen1107

**Comment by yifan-gu** [comment]
LGTM

**Comment by dchen1107** [comment]
cc/ @kubernetes/goog-node

**Comment by dchen1107** [comment]
ok to test

**Comment by dchen1107** [comment]
LGTM

**Comment by yujuhong** [comment]
LGTM

File: pkg/kubelet/image_manager_test.go [hunk 1]
```diff
@@ -246,53 +275,51 @@ func TestFreeSpaceRemoveByLeastRecentlyUsed(t *testing.T) {
...
-			ID:          "5678",
-			RepoTags:    []string{"potato", "salad"},
-			VirtualSize: 2048,
+			ID:   "5678",
+			Tags: []string{"potato", "salad"},
+			Size: 2048,
 		},
 	}
-	fakeDocker.ContainerList = []docker.APIContainers{
+	fakeRuntime.ContainerList = []*container.Container{
```
Comment (yujuhong) [inline pkg/kubelet/image_manager_test.go]: This should be fakeRuntime.PodList. The ContainerList is not used by the fakeRuntime at this moment. I will send a PR to remove that.

Comment (feiskyer) [inline pkg/kubelet/image_manager_test.go]: @yujuhong Thanks

**Pull request merged**